CMakeLists.txt

```diff
@@ -29,7 +29,7 @@ project (MelodyExtraction)
 ###############################################################################
 # C compiler flags
 ###############################################################################
-SET(CMAKE_C_FLAGS, "-Wall -O2 -g")
+SET(CMAKE_C_FLAGS, "-Wall -O3 -g")
 
 
```

Comment:
I would considering adding something like the following so that a macro to like `BIG_ENDIAN` get's defined on Big-Endian platforms.
```
###############################################################################
# Check endianness
###############################################################################
INCLUDE(TestBigEndian)
TEST_BIG_ENDIAN(big_endian)
if(${big_endian})
  # If the BIG_ENDIAN macro the source code assumes little endianness.
  add_definitions(-DBIG_ENDIAN)
endif()
```